packages/cpt-ui/components/EpsRoleSelectionPage.tsx

```diff
@@ -81,90 +81,96 @@ export default function RoleSelectionPage({ contentText }: RoleSelectionPageProp
     const [redirecting, setRedirecting] = useState<boolean>(false)
     const [rolesWithAccess, setRolesWithAccess] = useState<RolesWithAccessProps[]>([])
     const [rolesWithoutAccess, setRolesWithoutAccess] = useState<RolesWithoutAccessProps[]>([])
+    const [currentlySelectedRole, setCurrentlySelectedRole] = useState<RoleDetails | undefined>(undefined)
 
     const router = useRouter()
     const auth = useContext(AuthContext)
 
-    useEffect(() => {
-        setLoading(true);
-        setError(null);
-        setRolesWithAccess([]);
-        setRolesWithoutAccess([]);
-
-        if (!auth?.isSignedIn || !auth?.idToken) {
-            setError(null);
-            return;
-        }
-        // Now that we know there is an id token, check that it has a toString property.
-        // For some reason, it doesn't have this immediately, it gets added after a brief pause.
-        if (!auth?.idToken.hasOwnProperty('toString')) {
-            setError(null);
-            return;
+    const loginInfoMessage = currentlySelectedRole
```

Comment:
This feels like it should have a `useEffect` to update it if the `currentlySelectedRole` changes?